decode.go

```diff
@@ -1042,7 +1037,7 @@ func (d *Decoder) decodeStruct(ctx context.Context, dst reflect.Value, src ast.N
 		return errors.Wrapf(foundErr, "failed to decode value")
 	}
 
-	if len(unknownFields) != 0 && d.disallowUnknownField {
+	if len(unknownFields) != 0 && d.disallowUnknownField && src.GetToken() != nil {
```

Comment:
Why is this added comparison ( `src.GetToken() != nil` ) necessary?

Reply:
Before, when parsing inline structs, this block would generate unknown field errors for any fields that don't belong to the current inline struct, and then the removed block above would swallow those errors. By adding this comparison, we skip checking for unknown fields so that it can move on and perform the validation check below (since inline structs get parsed without a token).

Reply:
I understood ! Thank you.

Could you please put the comment like the following ?
`If there are unkonwn fields in the embedded struct by inline, the token does not exist, so ignore this section`

Reply:
Done (I reworded it to try to make it even clearer).